toolkit/tools/pkg/imagecustomizerlib/liveosisobuilder.go

```diff
@@ -580,7 +600,9 @@ func generatePxeGrubCfg(inputContentString string, pxeIsoImageBaseUrl string, px
 //     false otherwise.
 func containsGrubNoPrefix(filePaths []string) bool {
 	for _, filePath := range filePaths {
-		if filepath.Base(filePath) == grubx64NoPrefixBinary {
+		if runtime.GOARCH == "arm64" && filepath.Base(filePath) == grubAA64NoPrefixBinary {
```

Comment:
The function currently checks `runtime.GOARCH` on every loop iteration, which isn't necessary. I think `runtime.GOARCH` checking can be pulled out of the loop and just check file path in the loop, since the `filePath` is what iterated in the loop.